base/pkg/types.jl

```diff
@@ -40,15 +41,73 @@ show(io::IO, s::VersionSet) = join(io, s.intervals, " ∪ ")
 isempty(s::VersionSet) = all(isempty, s.intervals)
 in(v::VersionNumber, s::VersionSet) = any(i->in(v,i), s.intervals)
 function intersect(A::VersionSet, B::VersionSet)
-    ivals = vec([ intersect(a,b) for a in A.intervals, b in B.intervals ])
-    ivals = copy(ivals) # temporary bandaid for issue #4592
+    ivals = [intersect(a,b) for a in A.intervals for b in B.intervals]
```

Comment:
I'm getting
```
ERROR: LoadError: LoadError: ArgumentError: argument to Flatten must contain at least one iterator
 in start at .\iterator.jl:599
 in grow_to! at .\array.jl:357
 in intersect at .\pkg\types.jl:44
 in push! at .\pkg\types.jl:182
 in requirements at .\pkg\query.jl:34
 in resolve at .\pkg\entry.jl:476
```
when I try to backport this to release-0.5 - the old version with `vec` should be equivalent right?